examples/ex1.cpp

```diff
@@ -101,6 +105,12 @@ int main(int argc, char *argv[])
    //    CUDA, OCCA, RAJA and OpenMP based on command line options.
    Device device(device_config);
    device.Print();
+   if (algebraic_ceed)
+   {
+      MFEM_VERIFY(DeviceCanUseCeed(),
+                  "--algebraic makes no sense without Ceed backend!");
+      MFEM_VERIFY(pa, "--algebraic only makes sense with partial assembly");
+   }
```

Comment:
I wish this could be checked inside the `ceed::AlgebraicSolver`, to keep `ex1` as clean as possible.

Reply:
I moved the checks into the constructor, which makes `ex1` look nicer. One thing I don't like is now in some cases you can use `--algebraic` on the command line and it will just not do anything (for example without `-pa`), which might be confusing to users. I'm open to suggestions.

Reply:
It's a good point, however it's already the case with `-d ceed-cpu` for instance, but I agree that it would be nice to detect it in some way. BTW, shouldn't the algebraic solver work on any `CeedOperator`, not only the `-pa` ones? Have you tried on the `-mf` ones?

Reply:
I went ahead and tried with `AssemblyLevel::NONE` (commented line 937-938 in algebraic.cpp), and it seems to work just fine (I get the same number of iterations as with `AssemblyLevel::PARTIAL), could you check and make the change if you agree?

Reply:
This surprised me at first but I think you're right; whatever assembly level Ceed supports can also be used by the solver.